### ISSUE TITLE
Shortcode

### ISSUE THREAD

**Issue by komitaltrade** [mautic/mautic-wordpress#14]
Can you please enable also shortcode for Landing page created in Mautic, as it will be more than useful to be capable easy integration in (ie) WordPress site.

**Comment by escopecz** [comment]
That would be technically quite hard because if we embed it like we do with Mautic Forms (via JS), search engines will not be able to index it and the whole idea of the Landing Page will lost one important feature. If you'll be able to provide some technical details how to embed a Mautic Landing Page which will then be still index-able by search engines, we can discus it.

**Comment by komitaltrade** [comment]
First - thanks a lot for great project (I discover it just last week, as marketing just now become interesting area for me (despite my master degree in marketing). However, thanks to that my education, I realized huge ptential exact of this issue of Landing Page and I will try to be brief.

P.S. I'm working on integrated development service pakage(s) and must to note that I will get integrated also PIWIK (where I can get A LOT OF DATA + MOBILE ACCESS).

TECHICAL POINT

Mautic can (should) to index (track) CSS ID and/or CSS CLASS (Mautic Landing PAge - Builder - Element - Advanced Tab - ID and/or Stylesheet Classes). PIWIK certanly will do it. 

P.S. So, if Mautic cannot use the those data (ID and CLASS), client can track elements in some level trough the PIWIK (if LP will be integrated via shortcode).

MARKETING POINT

It is wrong point of view that only Campaign "nest" LP(s) and "vice versa" is useless, as facts shows contrary. Well done LP contain at least several Campaigns and great done LP is ENTIRELY built from Campaigns Elements.

Idea that LP contain just one Element (by rule - considered is only CTA button - wrong) is by definition huge error, as many visitors needs before decission more informations ("keep reading", "reference links", "testimonials", "contact form", ... buttons, as well as sections).

So, if client can follow Elements (visitor Actions) "success", he can significantly improve his further LP(s) concept, look & feel, content, "flow order", ....

PRAGMATIC POINT

Why tracking should to be condition???

Don't you think that painless integration of Mautic in one (ie) WP site (visually, but also in substance) is not from great benefit to Mautic future???

I intent (ie) to deliver to my clients LP template with integrted Mautic created LP (they can painless to add, change, remove, ... from WP site). They just need in Mautic Dashboard to create it and than to (re)assign ID in WP page shortcode and new LP can be by client itself to dinamically "work" for client needs.

In other words, created (followed) Campaigns in Mautic Dashboard (by client), as well as integrated (tracked) forms can get the full sense anly if client can dinamically to work with LP part of his entire Website, without need to "mess" with main Website.

Is it pragmatic approach?

**Comment by escopecz** [comment]
TBH, I don't understand any of your points. Partly because of acronyms used, partly because English is my second language probably. 

I really don't see the point in embedding a Mautic Landing Page into CMS. When you use CMS, don't use Mautic Landing Pages. Why would you? Use this plugin and Mautic will track your WordPress for you like it would be a Mautic Landing Page.

**Comment by komitaltrade** [comment]
Simple. You are developer and you don't think about the ordinary client. If they have embeded Mautic LP, they don't need to open and "mess" with WP Dashboard to change LP per Campaign needs, they can only to use Mautic Dashboard and to change content of WP site by change of Mautic LP as content of WP LP will be changed because it is not the WP LP - it is embeded Mautic LP.

Is it now clear?

**Comment by dbhurley** [comment]
I understand what you are suggesting. I would be interested in seeing if others in the community also saw this as a need. Thank you for mentioning it!

**Comment by escopecz** [comment]
Yeah, now I understand. And I agree it would be easier for the clients to use one system for everything. The idea right now is to use the right system for the right task. CRM for customer relationship management, CMS for content management, PMS to manage projects, Mautic for marketing automation. All those systems can be connected and communicate between each other and Mautic has quite a lot of plugins to make it simple. One day, maybe, we'll get to the point where all tasks could be done via Mautic, but for now, Mautic is primarily for marketing automation.

I'm still not sure how the embed would technically work though.

Anyway, it is a good idea.

**Comment by komitaltrade** [comment]
Please don't forget that many aspects what Mautic not track (analize), user can do trough other tools (I indicated PIWIK), what can add additional value of Mautic integrated with WP (in this case).

However, I'm quite simple in thoughts. Practical example.

I did and I host on WPMU (AWS hosted) hotel domain entirely done by WP. My clients are interested to change dinamically their LP (and to track results) by seasons, holidays, events, package discount campaigns, .... and they would like to not EVER OPEN WP Dashboard, rather to simply by drag and drop Mautic (simple but effective) builder, to change packages (content, price, ...), as well the content (surf season, whale watching season, carnaval season, ...), without paying to me development changes on main WP (they want to stick on monthly paymant of clear hosting with basic maintaince tasks - as campaign changes will be charged separatelly).

That is more than AMAZING benefit of embeded LP.

BTW - if you can embed in WP Mautic Form Page (MFP) via shortcode, what is the real problem to do it with MLP???

**Comment by escopecz** [comment]
Yes, the visitor action tracking is something (added to cart, a product paid, ...) I'd like to see in Mautic as well. 

I described the problem with embedding Mautic Landing Page the same way as Mautic Forms in my first comment.

**Comment by komitaltrade** [comment]
Yes, you wrote "...if we embed it like we do with Mautic Forms (via JS), search engines will not be able to index it...", but there is not described what is by your point of view the technical problem to embed the page. That is expalnation that you cannot embed the indexing (BTW - why you can not embeed/index ID and classses and you can embeed form fields???).

I just above tried to point that also if you cannot solve the indexing, that is not reason to reject other benefits of option to easy embed the MLP in WP. That is (as far as I know) two COMPLETE separate issues (embeding and indexing). You can enable embed without the option to enable "embeded indexing" (or I'm wrong?).

**Comment by escopecz** [comment]
You'd use a web page which will not be indexed by the search engines? I wouldn't. From my point of view, it is useless to have a website when no one will know about and therefore no one will visit. Unless I pay for the advertisement.

Let's look at how the forms are embedded technically:
1. The parent page is loaded.
2. Browser starts to execute JS on that page.
3. Forms are loaded asynchronously by the JS.

Now, search engine bots which are indexing the web pages see only the page in the state of 1. They do not execute the JS and do not wait what content will appear after.

In case of forms, they doesn't need to be indexed because there is almost no text to be indexed in them. That is not the case of Landing Pages though.

Id you do not care about the search engines, you can use Mautic Landing Page embed today. Use iframe for it.

**Comment by komitaltrade** [comment]
Sorry, now I get your concern and here is answer.

MLP is EMBEDED in WP page, what obviously mean that WP page is indexed with entire (embeded) content (I guess as well with content from MLP).

So, maybe MLP content will not be indexed, but WP page will be, what however cannot meke HUGE diference.

**Comment by komitaltrade** [comment]
iFrame
NOPE, as client cannot manipulat with it in WP (they should/want to change MLP, remember? - the ID will be changed).

**Comment by AndreaBarghigiani** [comment]
I've seen this kind of feature in a similar product that's called LeadPages, they've allow their user to include a landing page into the CMS with a different approach than shortcodes.

Here is the documentation: https://support.leadpages.net/hc/en-us/articles/203522080-Installing-the-WordPress-Plugin-version-2-0-

I think that with Mautic we can do something similar because WordPress allow users to set their Template Page and in this way you can override all the template engine and let WordPress load only your code.

Hope this helps and I am looking to discussing it even further.

**Comment by shulard** [comment]
Hello,

Thank you @AndreaBarghigiani for the link. It's a bit late but can you explain this different approach here ?

Maybe embedding a LP inside a WordPress project can be done in a different way than JS :

* Make server API call to retrieve details and build result before the page was sent to browser ;
* Retrieve some of the LP details using API (for SEO) and let JS load asynchronously all the others things...

**Comment by shulard** [comment]
I close the issue since there is no activity.

**Comment by shulard** [comment]
Since we are going to introduce a Mautic API Wrapper in the plugin (see #62), I think we'll have all the required thing to retrieve LP details and embed it (using shortcode or anything else...).
